.github/workflows/test-suite.yml

```diff
@@ -252,6 +252,23 @@ jobs:
       run: make lint
     - name: Certify Cargo.lock freshness
       run: git diff --exit-code Cargo.lock
+  disallowed-from-async-lint:
+    name: disallowed-from-async-lint
+    runs-on: ubuntu-latest
+    needs: cargo-fmt
+    continue-on-error: true
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install SigP Clippy fork
+      run: |
+        cd ..
+        git clone https://github.com/michaelsproul/rust-clippy.git
+        cd rust-clippy
+        git checkout 31a49666ccfcd7963b63345d6ce757c373f22c2a
+        cargo build --release --bin cargo-clippy --bin clippy-driver
+        cargo build --release --bin cargo-clippy --bin clippy-driver -Zunstable-options --out-dir $(rustc --print=sysroot)/bin
```

Comment:
We build twice intentionally: the first time downloads the toolchain and builds the binaries, the second time copies the built binaries to the correct path (using the `rustc` that was downloaded on the first run).